### PR TITLE
Migration to Swan Lake Alpha2 - Azure Storage Service - File module

### PR DESCRIPTION
**Pull request by SanduDS**
Resolves https://github.com/wso2-enterprise/choreo/issues/2739

## Automation tests
 - Unit tests 
   > Unit tests done 
 - Integration tests
   > Done and verified

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

**Comment by anupama-pathirage** [comment]
@SanduDS The commit messages and PR heading is not according to the best practices. Please correct them.

A properly formed Git commit subject line should always be able to complete the following sentence:

If applied, this commit will **_your subject line here_**

Refer [1] for more details.

[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md#submit-your-contribution

File: modules/files/serviceLevel_endpoint.bal [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    private http:Client httpClient;
+    private boolean isSharedKeyUsed = false;
+    private AzureConfiguration azureConfig;
+
+    # Initalize Azure Client using the provided azureConfiguration by user
+    #
+    # + azureConfig - AzureConfiguration record
+    public function init(AzureConfiguration azureConfig) returns error? {
+        http:ClientSecureSocket? secureSocketConfig = azureConfig?.secureSocketConfig;
+        self.sharedKeyOrSASToken = stringLib:substring(azureConfig.sharedKeyOrSASToken, startIndex = 1);
```
Comment (LakshanSS) [inline modules/files/serviceLevel_endpoint.bal:33]: Can be simplified to

self.sharedKeyOrSASToken = azureConfig.sharedKeyOrSASToken.substring(startIndex, endIndex);

File: modules/files/utils.bal [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+                    };
+                    prepareAuthorizationHeaders(authorizationDetail);    
+                } else {
+                    if (isFirstRequest){
+                        string tokenWithAmphasand = stringLib:concat(AMPERSAND, stringLib:substring(
+                            azureConfig.sharedKeyOrSASToken, startIndex = 1));
+                        requestPath = stringLib:concat(requestPath, tokenWithAmphasand);
+                        isFirstRequest = false;
+                    } 
+
```
Comment (LakshanSS) [inline modules/files/utils.bal:352]: Can remove the new line

**Comment by LakshanSS** [comment]
@SanduDS We can close this PR since this is done via https://github.com/ballerina-platform/module-ballerinax-azure-storage-service/pull/4

**Comment by SanduDS** [comment]
This is closed as it is outdated by #4

**Pull request closed without merging**